gstools/field/tools.py

```diff
@@ -0,0 +1,269 @@
+    if np.size(func_val) > 1:
+        return list_format(func_val, prec=f_cls.model._prec)
+    return "{0:.{p}}".format(float(func_val), p=f_cls.model._prec)
+
+
+def _fmt_normalizer(f_cls):  # pragma: no cover
+    norm = f_cls.normalizer
+    return str(None) if norm.__class__ is Normalizer else norm.name
+
+
+def fmt_mean_norm_trend(f_cls):  # pragma: no cover
+    """Format string repr. for mean, normalizer and trend of a field."""
+    args = [
+        "mean=" + _fmt_func_val(f_cls, f_cls.mean),
+        "normalizer=" + _fmt_normalizer(f_cls),
+        "trend=" + _fmt_func_val(f_cls, f_cls.trend),
+    ]
+    return "".join([", " + arg for arg in args if not arg.endswith("None")])
+
+
+def to_vtk_helper(
+    f_cls, filename=None, field_select="field", fieldname="field"
+):  # pragma: no cover
+    """Create a VTK/PyVista grid of the field or save it as a VTK file.
+
+    This is an internal helper that will handle saving or creating objects
+
+    Parameters
+    ----------
+    f_cls : :any:`Field`
+        Field class in use.
+    filename : :class:`str`
+        Filename of the file to be saved, including the path. Note that an
+        ending (.vtr or .vtu) will be added to the name. If ``None`` is
+        passed, a PyVista dataset of the appropriate type will be returned.
+    field_select : :class:`str`, optional
+        Field that should be stored. Can be:
+        "field", "raw_field", "krige_field", "err_field" or "krige_var".
+        Default: "field"
+    fieldname : :class:`str`, optional
+        Name of the field in the VTK file. Default: "field"
+    """
+    if f_cls.value_type == "vector":
+        if hasattr(f_cls, field_select):
+            field = getattr(f_cls, field_select)
+        else:
+            field = None
+        if not (f_cls.pos is None or field is None or f_cls.mesh_type is None):
+            suf = ["_X", "_Y", "_Z"]
+            fields = {}
+            for i in range(f_cls.model.dim):
+                fields[fieldname + suf[i]] = field[i]
+            if filename is None:
+                return to_vtk(f_cls.pos, fields, f_cls.mesh_type)
+            else:
+                return vtk_export(filename, f_cls.pos, fields, f_cls.mesh_type)
+    elif f_cls.value_type == "scalar":
+        if hasattr(f_cls, field_select):
+            field = getattr(f_cls, field_select)
+        else:
+            field = None
+        if not (f_cls.pos is None or field is None or f_cls.mesh_type is None):
+            if filename is None:
+                return to_vtk(f_cls.pos, {fieldname: field}, f_cls.mesh_type)
+            else:
+                return vtk_export(
+                    filename, f_cls.pos, {fieldname: field}, f_cls.mesh_type
+                )
+        else:
+            print("Field.to_vtk: '{}' not available.".format(field_select))
+    else:
+        raise ValueError(
+            "Unknown field value type: {}".format(f_cls.value_type)
+        )
+
+
+def mesh_call(
```

Comment:
I think I don't like the name of the function.

Reply:
renamed to `generate_on_mesh`.